carRental/src/main/java/com/amigoscode/chohort2/carRental/carProviderCoupon/CarProviderCouponService.java

```diff
@@ -27,6 +31,20 @@ public CarProviderCouponDTO save(CarProviderCouponVM carProviderCouponVM) {
         return CarProviderCouponMapper.INSTANCE.toDto(carProviderCouponRepository.save(carProviderCoupon));
     }
 
+    public CarProviderCouponDTO updateCouponAvailability(Long id,CarProviderCouponAvailabilityVM carProviderCouponAvailabilityVM){
+          CarProviderCoupon carProviderCoupon = getCarProviderCouponById(id);
+          if(carProviderCoupon.getCarProviderId().equals(getCarProvideId())) {
+              carProviderCoupon.setIsAvailable(carProviderCouponAvailabilityVM.getIsAvailable());
+          }
+        return CarProviderCouponMapper.INSTANCE.toDto(carProviderCouponRepository.save(carProviderCoupon));
+    }
+
+    private CarProviderCoupon getCarProviderCouponById(Long id){
+        CarProviderCoupon carProviderCoupon = carProviderCouponRepository.findById(id)
+                .orElseThrow(()->new ApiRequestException("Coupon not found"));
+        return carProviderCoupon;
+    }
+
```

Comment:
Maybe it would be better if the method's name was get**LoggedIn**CarProviderId() or get**Current**CarProviderId(). Perhaps.